File: OffByOne/src/OffByOne.Qls/Ast/Style/Widgets/Base/Widget.cs
namespace OffByOne.Qls.Ast.Style.Widgets.Base
{
    using OffByOne.Ql.Ast;
    using OffByOne.Ql.Common.Visitors.Contracts;
    using OffByOne.Qls.Common.Visitors.Contracts;

    public abstract class Widget : AstNode, IVisitableWidget
    {
        public abstract TResult Accept<TResult, TContext>(IWidigetVisitor<TResult, TContext> visitor, TContext environment)
            where TContext : IEnvironment;
    }
}

File: Loading/QLS/src/qls/ast/widget/Widget.java
package qls.ast.widget;

import QL.ast.Node;
import QL.ast.type.Type;
import QL.ui.field.Field;
import qls.evaluation.Evaluator;

public abstract class Widget extends Node {

	Widget(int line) {
		super(line);	
	}

	public abstract Type getType();
	
	public abstract Field accept(Evaluator v); 

}
